sphinx_rtd_theme/layout.html

```diff
@@ -133,23 +135,32 @@
         <div class="wy-side-nav-search" {% if theme_style_nav_header_background %} style="background: {{theme_style_nav_header_background}}" {% endif %}>
           {%- block sidebartitle %}
 
-          {%- if logo and theme_logo_only %}
+          {%- if ( logo or logo_url ) and theme_logo_only %}
+
+          {%- if sphinx_version_info < (4, 0) -%}
```

Comment:
This logic is a bit hard to follow, and I suspect a conditional on the Sphinx version is not needed. The following scenarios are possible:

* Sphinx < 4 + logo
* Sphinx 4<x<6 + logo -- this one seems missing from some of the logic below
* Sphinx 4<x<6 + logo_url

Instead, this decision tree can be reduced if we just set the variables we use ahead of time:

```
{% set _logo_url = logo_url|default:pathto('_static/' + logo, 1) %}
{% set _root_doc = root_doc|default:master_doc %}
```

Then just use those variables instead of using conditional logic in the templates. Does that have the same effect you're aiming for?

Reply:
I understand that it's possible to condense this into shorter statements, but I preferred the explicit statements that express version logic. It makes it easier to spot what may be eliminated once support of some version is dropped.

Or that was the intention.

If you prefer condensed expressions, I can rewrite it and test it again.

Reply:
It's not as much about shortening this as it is about preserving the existing logic. The change here seems to drop support for `Sphinx 4<x<6` and usage of the `logo` variable. This is still valid usage, Sphinx 4 and 5 have both `logo` and `logo_url`. Relying on the Sphinx version in this logic is making it easier to break this conditional.

In the end, I don't think we care about what version of Sphinx the user is using, just what variables they are using.

Reply:
I know what you mean because that was also my first implementation draft until I realized that `logo_url` will exist always if `logo` is set. It's implemented like this in Sphinx 4-6:

https://github.com/sphinx-doc/sphinx/blob/c1f2a52d5d97f874d84e4d5421c99a566242af6b/sphinx/builders/html/__init__.py#L1252-L1257

Reply:
[Mentioned below](https://github.com/readthedocs/sphinx_rtd_theme/pull/1405/files#r1068646958), I had to look at Sphinx's code to verify it was possible to use `logo` still. This at very least is worth noting in the logic.

Though I still feel this would be far clearer using variables instead of nested conditionals

Reply:
@agjohnson `html_logo` is a configuration value: https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_logo and `logo_url` is the only context variable since Sphinx 6.

I've implemented the condensed versions, removing some Django template syntax -- I know you love it deep down inside :smile: